Hunting Queries/SecurityEvent/FirewallRuleChanges_using_netsh.yaml

```diff
@@ -0,0 +1,30 @@
+id: 3dc5dc8b-160b-407e-9925-24a91e3599df
+name: Firewall rule changes using netsh
+description: |
+  This query will show firewall rule changes using netsh utility.
+  This technique was seen in relation to Solarigate attack but the results can indicate potential  malicious activity used in different attacks.
+  Refer to netsh syntax: https://docs.microsoft.com/windows-server/administration/windows-commands/netsh
+severity: Low
+requiredDataConnectors:
+  - connectorId: SecurityEvents
+    dataTypes:
+      - SecurityEvent
+tactics:
+  - Execution
+relevantTechniques:
+  - T1204
+query: |
+  let timeframe = 1d;
+  let AccountAllowList = dynamic(['SYSTEM']);
+  let tokens = dynamic(["add", "delete", "set"]); 
+  SecurityEvent
+  | where TimeGenerated >= ago(timeframe)
+  | where Process =~ "netsh.exe" 
+  | where CommandLine has_all ("advfirewall", "firewall") and CommandLine has_any (tokens)
```

Comment:
This alone will likely provide FPs.  I suggest we take a look at common usages and remove them via dynamic list or we use stricter match method.  For example, this will get a hit on SQL Server (BKUPEXEC).  Using a dynamic list may be fragile, so have a look at the command line identified in our blog a bit further to see what we could do here. https://www.microsoft.com/security/blog/2021/01/20/deep-dive-into-the-solorigate-second-stage-activation-from-sunburst-to-teardrop-and-raindrop/

Reply:
Working on rewriting this query with Ashwin to take into account rule name and possibly other options.